### PR TITLE
Create M6: Draft-Insufficient-Input-Output-Validation.md

### PR DESCRIPTION
**Pull request by kunwaratulhax0r**

### CODE REVIEW

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+title: "M6: Insufficient Input/Output Validation"
+---
+
+# Threat Agents
+
+**Application Specific**
+
+If the application fails to properly validate and sanitize data received from external sources like user inputs or network data. This oversight can lead to various security vulnerabilities, such as SQL injection, Command Injection or cross-site scripting (XSS) attacks. 
+
+These attacks can lead to unauthorized data access, manipulation, and potentially compromise the entire system.
```
Comment (3asm) [inline 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:13]: Maybe replace the entire system with something mobile-specific. This wording seems like it is talking about backends.

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+These attacks can lead to unauthorized data access, manipulation, and potentially compromise the entire system. 
+
+Inadequate output validation can result in data corruption or presentation vulnerabilities, allowing malicious actors to inject malicious code or manipulate sensitive information displayed to users.
+
+# Attack Vectors	
+
+**Exploitability HIGH**
+
+Insufficient input/output validation exposes our application to critical attack vectors, including SQL injection, XSS, command injection, path traversal, HTML injection, and remote file inclusion. These vulnerabilities can lead to unauthorized access, data manipulation, code execution, and compromise of the entire system.
```
Comment (3asm) [inline 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:21]: I would remove remote file inclusion which is very rare on the client side due to the languages commonly in use in mobile apps.

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+- Data Breaches: Insufficient validation allows attackers to manipulate input, resulting in data breaches and unauthorized access to sensitive information.
+- System Disruptions: Exploitation of the vulnerability can cause application crashes, instability, or data corruption, leading to service disruptions and operational inefficiencies.
+- Data Integrity Issues: Insufficient validation may result in data corruption, incorrect processing, or inaccurate outputs, compromising the reliability and integrity of the system.
+
+On the business side, the impacts include:
+- Reputation Damage: Successful exploitation of the vulnerability can result in data breaches, system disruptions, and customer distrust, damaging the organization's reputation and brand image.
+- Legal and Compliance Consequences: Non-compliance with data protection regulations due to insufficient validation can lead to legal liabilities, regulatory penalties, and potential financial losses.
+- Financial Impact: Data breaches or system disruptions caused by the vulnerability can result in financial losses due to incident response, remediation costs, legal fees, and potential loss of revenue.
```
Comment (3asm) [inline 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:73]:
```suggestion
Insufficient input/output validation vulnerability has significant technical and business implications. From an application standpoint, the impacts include:

- Code Execution: Attackers can exploit this vulnerability to execute unauthorized code, potentially leading to system compromise and unauthorized access.
- Data Breaches: Insufficient validation allows attackers to manipulate input, resulting in data breaches and unauthorized access to sensitive information.
- System Disruptions: Exploitation of the vulnerability can cause application crashes, instability, or data corruption, leading to service disruptions and operational inefficiencies.
- Data Integrity Issues: Insufficient validation may result in data corruption, incorrect processing, or inaccurate outputs, compromising the reliability and integrity of the system.

On the business side, the impacts include:
- Reputation Damage: Successful exploitation of the vulnerability can result in data breaches, system disruptions, and customer distrust, damaging the organization's reputation and brand image.
- Legal and Compliance Consequences: Non-compliance with data protection regulations due to insufficient validation can lead to legal liabilities, regulatory penalties, and potential financial losses.
- Financial Impact: Data breaches or system disruptions caused by the vulnerability can result in financial losses due to incident response, remediation costs, legal fees, and potential loss of revenue.
```

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+**Data Breaches:** Insufficient validation can enables attackers to manipulate input, potentially leading to unauthorized access and extraction of sensitive data.
+
+**System Compromise:** Attackers can gain unauthorized access to the underlying system, compromising it and potentially taking control.
+
+**Application Disruption:** Malicious input can cause disruptions, crashes or data corruption, impacting the application's reliability and functionality.
+
+**Reputation Damage:** Successful exploitation of this vulnerability can result in reputational harm due to data breaches and loss of customer's trust.
+
+**Legal and Compliance Issues:** Inadequate validation may lead to legal liabilities, regulatory penalties and non compliance with data protection regulations.
```
Comment (3asm) [inline 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:56]:
```suggestion
Insufficient input/output validation vulnerability can have several technical impacts on the affected application:

**Code Execution:** A malicious actor can exploit this vulnerability to execute unauthorized code within the application's environment, bypassing the security measures.

**Data Breaches:** Insufficient validation can enable attackers to manipulate input, potentially leading to unauthorized access and extraction of sensitive data.

**System Compromise:** Attackers can gain unauthorized access to the underlying system, compromising it and potentially taking control.

**Application Disruption:** Malicious input can cause disruptions, crashes, or data corruption, impacting the application's reliability and functionality.

**Reputation Damage:** Successful exploitation of this vulnerability can result in reputational harm due to data breaches and loss of customer trust.

**Legal and Compliance Issues:** Inadequate validation may lead to legal liabilities, regulatory penalties, and non-compliance with data protection regulations.
```

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+ 
+# Example Attack Scenarios
+
+**Scenario #1** SQL Injection
+
+An attacker can craft malicious input by appending SQL statements to user-provided data. For example, in a login form, an attacker could enter a specially crafted username like "' OR 1=1;--" which, if the application lacks proper input validation, can alter the intended SQL query. This can lead to the execution of unauthorized database operations, bypassing authentication, and potentially accessing sensitive information or gaining administrative privileges.
+
+**Scenario #2** Cross Site Scripting
+
+An attacker inject malicious scripts into input fields such as comment sections or form inputs. When the application fails to adequately validate or sanitize this output data, the injected scripts are executed by unsuspecting users' browsers. This enables the attacker to steal session cookies, perform actions on behalf of the victim, or deface the website.
```
Comment (3asm) [inline 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:111]: This is very web specific, need a different example.

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+- Secure Coding Practices:
+  - Follow secure coding practices, such as using parameterized queries and prepared statements to prevent SQL injection.
+- Regular Security Testing:
+  - Conduct regular security assessments, including penetration testing and code reviews, to identify and address vulnerabilities.
+ 
+# Example Attack Scenarios
+
+**Scenario #1** SQL Injection
+
+An attacker can craft malicious input by appending SQL statements to user-provided data. For example, in a login form, an attacker could enter a specially crafted username like "' OR 1=1;--" which, if the application lacks proper input validation, can alter the intended SQL query. This can lead to the execution of unauthorized database operations, bypassing authentication, and potentially accessing sensitive information or gaining administrative privileges.
```
Comment (3asm) [inline 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:107]: Need more Mobile nuances, like attack Providers.

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+
+**Scenario #2** Cross Site Scripting
+
+An attacker inject malicious scripts into input fields such as comment sections or form inputs. When the application fails to adequately validate or sanitize this output data, the injected scripts are executed by unsuspecting users' browsers. This enables the attacker to steal session cookies, perform actions on behalf of the victim, or deface the website.
+
+**Scenario #3** Command Injection
+
+An Attacker can exploit insufficient input validation by injecting malicious system commands into input fields. For example, an attacker might append a command like "; rm -rf /" to a user-provided filename, which, if the application does not properly validate the input, could result in the execution of the malicious command. This can lead to unauthorized access, data loss, or complete compromise of the underlying system.
```
Comment (3asm) [inline 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:115]: Also very web specific.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 1 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:13]: Yes, backend system or maybe we can replace it with Database?

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 2 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:21]: ok, will do

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 3 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:73]: noted.

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 4 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:56]: noted

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 5 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:111]: sure

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 6 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:107]: sure

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 7 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md:115]: will look into it

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 1 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md]: review pls.

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 5 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md]: made some changes.

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 6 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md]: made some changes.

File: 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md (hunk 7 above)
Comment (kunwaratulhax0r) [reply 2023-risks/M6: Draft-Insufficient-Input-Output-Validation.md]: made some changes.

**Pull request merged**